### DISCUSSION TITLE
How to disable the integrated Comment system?

### DISCUSSION THREAD

**Discussion by andrey-jef** [ObsidianPublisher/obsidian-github-publisher#70] [Q&A — Plugin] [answered]
Hi Mara,

Following [this tutorial from mkdocs-material](https://squidfunk.github.io/mkdocs-material/setup/adding-a-comment-system/), "you can enable comments on a page by setting the comments front matter property to true". 

I'm understanding that if the frontmatter does not included the `comment` keyword, then the comments will be disabled, is it correct?

However, in my recent try, the Comments section was always there (check screenshot below)

Could you please teach me how to 
- disable the Comments for all pages?
- enable the Comments for only some specified pages, and disable for the whole remaining?

I already read [the tutorial](https://obsidian-publisher.netlify.app/setup/customization/#comments) to integrate the Comments in the plugin documentation site, but I could not find how to disable. 

At the moment, I incline to delete all the contents in the file `/overrides/partials/comments.html` or even delete the file `comments.html`. But I'm afraid to change anything which can break the published site.

Thanks

![Screenshot 2023-01-11 051613](https://user-images.githubusercontent.com/81280428/211716598-f62abbe8-c75c-42f4-b6e7-2b95741d4300.jpg)

**Comment by Mara-Li** [comment]
Hello!
Sorry, I don't know how to write documentation :'D

You can disable it : 
- Globally with deleting the `overrides/partials/comments.html` and removing this part in the `main.html` : 
	```
	{% block content %}
	  {{ super() }}
	  {% include "partials/comments.html" %}
	{% endblock %}
	```
- To enable in specific page, create a template as follow : 
	```
	{% extends "base.html" %}
	{% block extrahead %}
		<script src="https://unpkg.com/@popperjs/core@2/dist/umd/popper.min.js"></script>
		<script src="https://unpkg.com/tippy.js@6/dist/tippy-bundle.umd.js"></script>
		<link
			rel="stylesheet"
			href="https://unpkg.com/tippy.js@6/animations/scale-subtle.css"
	  	/>
	    <link
			rel="stylesheet"
			href="https://unpkg.com/tippy.js@6/themes/translucent.css"
	    />
		<link
		    rel="stylesheet"
		    href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0-beta3/css/all.min.css"
		    integrity="sha512-Fo3rlrZj/k7ujTnHg4CGR2D7kSs0v4LLanw2qksYuRlEzO+tcaEPQogQ0KaoGN26/zrn20ImR1DfuLWnOo7aBA=="
		    crossorigin="anonymous"
		    referrerpolicy="no-referrer"
		/>
  		{% include "partials/meta.html" %}
	{% endblock %}
	{% block content %}
	  	{{ super() }}
	  	{% include "partials/comments.html" %}
	{% endblock %}
	```
	Alternatively, you can juste make a copy of the actual `main.html` and rename it. 
	After that, in page you want to enable the comments, you must add the key `template: file.html` in the frontmatter.
	ie, if you name it `page_comments.html`, you enable it with `template: page_comments.html`

**Reply by andrey-jef** [reply]
Thanks. Just a minor issue.

Here is what I have done to keep the `Comments` for only some specific pages, as your instructions:
- duplicate `main.html` and rename it to `page_comments.html`
- remove this part from `main.html`
```html
 {% block content %}
   {{ super() }}
   {% include "partials/comments.html" %}
 {% endblock %}
```
- then build and go check in a page where I did NOT include the key `template: page_comments.html` in the page's frontmatter

As in the screenshot, you will see that the `Comments` section is still there, in the page without the frontmatter key `template: page_comments.html`.

The `Comments` section only disappear totally when I remove the `partials/comments.html` file or rename it. But this action means that I have to disable the `Comments` globally, for all of pages.

Do you have another suggestion on how to keep the `Comments` partially, for some specific pages only?

![Screenshot 2023-01-12 013627](https://user-images.githubusercontent.com/81280428/211949877-5afa5487-e7c0-4e7e-ac3e-7461cba76e66.jpg)

**Comment by andrey-jef** [comment]
@Lisandra-dev 
Further, do you remember in which part of the template that you modified, that make this option of mkdocs-materials (enable comments on a page by setting the `comments` front matter property to `true` [^1]) does not work?

Nothing in haste, I'm just curious to consult the docs, compare the code and learn new things.
Thanks

[^1]: https://squidfunk.github.io/mkdocs-material/setup/adding-a-comment-system/

**Reply by Mara-Li** [reply]
Oh, thanks. I will update the template :)

**Comment by Mara-Li** [comment, marked as answer]
Okay, it takes me some times but I successfully updated the method to add comments.

First, I updated the python hooks file `on_env.py` because Jinja doesn't work as attended. I added the function `value_in_frontmatter` : 
```python
def value_in_frontmatter(key, metadata):
    """Check if a key exists in the frontmatter.

    Args:
        key (any): the key to check
        metadata (any): the frontmatter

    Returns:
        bool: true if exists
    """
    if key in metadata:
        return metadata[key]
    else:
        return None
```
and updated the on_env function : 
```python
def on_env(env, config, files, **kwargs):
    env.filters["value_in_frontmatter"] = value_in_frontmatter
    return env
```

In comments, I added this line before the comments block : 
```jinja
{% set local_comments = "comments" | value_in_frontmatter(page.meta) %}
{% set comments = config.extra["comments"] if local_comments is none else local_comments %}
{% if comments %}
.... 
{% endif %}
```

Following this, you need to add the key `comments` in your `mkdocs.yml`, in the extra part : 
```yaml
extra:
  comments: true # or false
```

This key allow to **enable** OR **disable** globally the comments. 
- If you set `comments: true` : In this case, if you want to disable comments for some page, you need to set `comments: false` in the frontmatter of these pages.
- If you set `comments: false` you need to set the frontmatter key `comments` on true.
